autogpt_platform/backend/backend/data/diagnostics.py

```diff
@@ -0,0 +1,1525 @@
+                    status=exec.executionStatus,
+                    created_at=exec.createdAt,
+                    started_at=exec.startedAt,
+                    queue_status=None,
+                )
+            )
+
+        return results
+    except Exception as e:
+        logger.error(f"Error getting orphaned execution details: {e}")
+        raise
+
+
+async def get_long_running_executions_details(
+    limit: int = 100, offset: int = 0
+) -> List[RunningExecutionDetail]:
+    """
+    Get detailed information about long-running executions (RUNNING status >24h).
+
+    Args:
+        limit: Maximum number of executions to return
+        offset: Number of executions to skip
+
+    Returns:
+        List of long-running RunningExecutionDetail objects
+    """
+    try:
+        # RUNNING executions older than 24 hours
+        cutoff = datetime.now(timezone.utc) - timedelta(hours=24)
+
+        executions = await AgentGraphExecution.prisma().find_many(
+            where={
+                "executionStatus": AgentExecutionStatus.RUNNING,
+                "createdAt": {"lt": cutoff},
+            },
+            include={
+                "AgentGraph": True,
+                "User": True,
+            },
+            take=limit,
+            skip=offset,
+            order={"createdAt": "asc"},  # Oldest first
+        )
+
+        results = []
+        for exec in executions:
+            results.append(
+                RunningExecutionDetail(
+                    execution_id=exec.id,
+                    graph_id=exec.agentGraphId,
+                    graph_name=(
+                        exec.AgentGraph.name
+                        if exec.AgentGraph and exec.AgentGraph.name
+                        else "Unknown"
+                    ),
+                    graph_version=exec.agentGraphVersion,
+                    user_id=exec.userId,
+                    user_email=exec.User.email if exec.User else None,
+                    status=exec.executionStatus,
+                    created_at=exec.createdAt,
+                    started_at=exec.startedAt,
+                    queue_status=None,
+                )
+            )
+
+        return results
+    except Exception as e:
+        logger.error(f"Error getting long-running execution details: {e}")
+        raise
+
+
+async def get_stuck_queued_executions_details(
+    limit: int = 100, offset: int = 0
+) -> List[RunningExecutionDetail]:
+    """
+    Get detailed information about stuck queued executions (QUEUED >1h, never started).
+
+    Args:
+        limit: Maximum number of executions to return
+        offset: Number of executions to skip
+
+    Returns:
+        List of stuck queued RunningExecutionDetail objects
+    """
+    try:
+        # QUEUED executions older than 1 hour that never started
+        one_hour_ago = datetime.now(timezone.utc) - timedelta(hours=1)
+
+        executions = await AgentGraphExecution.prisma().find_many(
```

Comment:
can we just add offset param on `get_graph_executions` and use it instead? so the timezone headache we'll only handle at one place